back-end/src/sciler/handler/handler.go

```diff
@@ -80,25 +80,99 @@ func (handler *Handler) onConnectionMsg(raw Message) {
 		logrus.Info("connection message received from: ", raw.DeviceID)
 		value, ok2 := contents["connection"]
 		if !ok2 || reflect.TypeOf(value).Kind() != reflect.Bool {
-			logrus.Error("received improperly structured connection message from device " + raw.DeviceID)
+			logrus.Warn("received improperly structured connection message from device " + raw.DeviceID)
+			return
 		}
 		device.Connection = value.(bool)
 		handler.Config.Devices[raw.DeviceID] = device
 		handler.SendStatus(raw.DeviceID)
 	}
 }
 
+// checkStatusType checks if the type of the status change is correct for the component
+func (handler *Handler) checkStatusType(device config.Device, v interface{}, k string) error {
+	valueType := reflect.TypeOf(v).Kind()
+	if inputType, ok := device.Input[k]; ok {
+		switch inputType {
+		case "string":
+			{
+				if valueType != reflect.String {
+					return fmt.Errorf("input type string expected but %sfound as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "boolean":
+			{
+				if valueType != reflect.Bool {
+					return fmt.Errorf("input type boolean expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "numeric":
+			{
+				if valueType != reflect.Int && valueType != reflect.Float64 {
+					return fmt.Errorf("input type numeric expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "array":
+			{
+				if valueType != reflect.Slice {
+					return fmt.Errorf("input type array/slice expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		default:
+			// todo custom types
+			return fmt.Errorf("custom types like: %s, are not yet implemented", inputType)
+		}
+	} else if output, ok2 := device.Output[k]; ok2 {
+		switch output.Type {
+		case "string":
+			{
+				if valueType != reflect.String {
+					return fmt.Errorf("input type string expected but %sfound as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "boolean":
+			{
+				if valueType != reflect.Bool {
+					return fmt.Errorf("input type boolean expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "numeric":
+			{
+				if valueType != reflect.Int && valueType != reflect.Float64 {
+					return fmt.Errorf("input type numeric expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		case "array":
+			{
+				if valueType != reflect.Slice {
+					return fmt.Errorf("input type array/slice expected but %s found as type of value %v for %s", valueType.String(), v, k)
+				}
+			}
+		default:
+			// todo custom types
+			return fmt.Errorf("custom types like: %s, are not yet implemented", inputType)
+		}
```

Comment:
Write a test for this case

Reply:
but its not implemented, we don't want the outcome to be an error so we shouldn't test for that